marketplace/deployer_util/config_helper.py

```diff
@@ -400,6 +408,38 @@ def type(self):
     return self._type
 
 
+class SchemaGcp:
+  """Accesses top level GCP constraints."""
+
+  def __init__(self, dictionary):
+    self._nodes = _maybe_get_and_apply(dictionary, 'nodes',
+                                       lambda v: SchemaNodes(v))
+
+  @property
+  def nodes(self):
+    return self._nodes
+
+
+class SchemaNodes:
+  """Accesses GKE cluster node constraints."""
+
+  def __init__(self, dictionary):
+    if 'requiredOauthScopes' in dictionary:
```

Comment:
`_required_oauth_scopes` wouldn't be set for the instance if this if condition is `false`.

Instead, do this:

```python
self._required_oauth_scopes = dictionary.get('requiredOauthScopes', [])
if not isinstance(self._required_oauth_scopes, ...)
  ...
for scope in self._required_oauth_scopes:
  ...
```